packages/react-scripts/package.json

```diff
@@ -54,7 +56,8 @@
   },
   "devDependencies": {
     "react": "^15.5.4",
-    "react-dom": "^15.5.4"
+    "react-dom": "^15.5.4",
+    "typescript": "^2.6.2"
```

Comment:
Adding `typescript` as dev dependency is not intended - we've moved it to be a peer dependency for this package (see below), and it is added as a dev dependency by the init script when creating a new project.

Reply:
I'll check it out, but Travis crashed if typescript wasn't installed and I tried to run the tests without the cache.

Internally ts-jest uses typescript as a peer-dependency, and the tests are written in typescript, so the package needs typescript as a dependency somehow. I think the reason it used to work was that Travis cached the typescript package.
Note that if something is both a devdependency and a peerDependency, it means that we'll need typescript installed, but the end-user won't have, and will instead use the peerDependency.

I'll research the travis issue a little bit more though, to make sure that we're not adding it needlessly.

Reply:
It's currently added via the `init` script when a new project gets created, thus it won't be installed and added as dev dependency outside of the "regular" initialization process. I'm not sure if any of the tests is heading this way, though, but it's not impossible.

Reply:
What do you mean here? It doesn't seem to me like the init script is copying the devDependencies over? Only the predefined ones in 'types'

Reply:
No, it executes a child process that runs `npm install -D` resp. `yarn add -D` with the required `@types` packages and `typescript` in the created project folder:
https://github.com/wmonk/create-react-app-typescript/blob/7149b3796c1c2d08606eb1656b628fc69cac7f3f/packages/react-scripts/scripts/init.js#L107-L126

But as mentioned above, this only reliably happens in the "regular" (i.e.: user based) workflow, and it is possible that the tests are not heading this way.